src/configure/configure.ts

```diff
@@ -2,41 +2,57 @@ const uuid = require('uuid/v4');
 import { AppServiceClient } from './clients/azure/appServiceClient';
 import { AzureDevOpsClient } from './clients/devOps/azureDevOpsClient';
 import { AzureDevOpsHelper } from './helper/devOps/azureDevOpsHelper';
-import { AzureTreeItem } from 'vscode-azureextensionui';
+import { AzureTreeItem, UserCancelledError } from 'vscode-azureextensionui';
 import { exit } from 'process';
 import { generateDevOpsProjectName } from './helper/commonHelper';
 import { GenericResource } from 'azure-arm-resource/lib/resource/models';
 import { GraphHelper } from './helper/graphHelper';
 import { LocalGitRepoHelper } from './helper/LocalGitRepoHelper';
-import { Messages } from './messages';
+import { Messages } from './resources/messages';
 import { QuickPickItem } from 'vscode';
 import { ServiceConnectionHelper } from './helper/devOps/serviceConnectionHelper';
 import { SourceOptions, RepositoryProvider, extensionVariables, WizardInputs, WebAppKind, PipelineTemplate, QuickPickItemWithData } from './model/models';
+import { TracePoints } from './resources/tracePoints';
+import { TelemetryKeys } from './resources/telemetryKeys';
 import * as path from 'path';
 import * as templateHelper from './helper/templateHelper';
 import * as utils from 'util';
 import * as vscode from 'vscode';
+import { TelemetryHelper } from './helper/telemetryHelper';
 
-export async function configurePipeline(node: any) {
+const Layer: string = 'configure';
+
+export async function configurePipeline(telemetryHelper: TelemetryHelper, node: AzureTreeItem) {
+    let startTime = Date.now();
```

Comment:
A more elegant way to doing time would be have a function called InvokeWithTimer which takes a function as argument - and then wraps it within start time and end time, and sets the telemetry property. caller of configurePipeline can use it.
However no need to do this if this place is the only occurrence where you are doing this